### ISSUE TITLE
 Impact of Scale Parameter on YOLO11m Detection Performance for Small Objects

### ISSUE THREAD

**Issue by ErnisMeshi** [ultralytics/ultralytics#18011]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am using the YOLO11m object detection model to detect relatively small objects in an indoor environment. Experimentally, I observed that the model often fails to detect the object in a 720p original image resolution. However, when the image is slightly zoomed in, the model detects it successfully.

Based on these empirical results, I reduced the scale parameter from 0.5 to 0.1 during training. My intuition is that the model might be learning to "see" the object as closer than it actually is, likely because the object is relatively small in the image.

When I evaluate the model using the COCO API, I get this graph (attached below). However, during live testing, I find that the model trained with scale=0.1 performs better at detecting the object than the model trained with the default parameters.
![image](https://github.com/user-attachments/assets/5123b854-7b7a-4658-b6b0-8ce1facbb402)

Could you confirm whether my intuition is correct? Specifically:

Is it plausible that reducing the scale parameter improves the model’s ability to detect small objects by preventing it from learning to "expect" objects at a closer distance?
Are there additional adjustments to the augmentation parameters or training process that could help further improve detection performance for small objects?

I would appreciate it if you could guide me on how to systematically address this issue or suggest any best practices that might apply to my scenario.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ErnisMeshi, thank you for your interest in Ultralytics 🚀! We appreciate your detailed explanation and the COCO evaluation graph you shared.

Regarding your observations, we recommend reviewing the [Docs](https://docs.ultralytics.com) for information on augmentation and training best practices. Notably, you may find the [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) helpful in optimizing your training process for small object detection.

To address your specific scenario:

1. If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug the issue effectively. This could include a small subset of your dataset along with the relevant training commands and settings.
2. If this is a ❓ Question about custom training or hyperparameter adjustments, please provide additional details such as:
   - Dataset examples (e.g., representative images of your small objects)
   - Training logs or metrics (e.g., mAP results, loss plots)
   - Any unique augmentation or training parameter changes you've made (e.g., setting `scale=0.1`).

In general, reducing the `scale` parameter can indeed influence the model's ability to handle objects at different sizes. We'll have an Ultralytics engineer review your findings and share guidance on whether your intuition aligns with best practices.

## Upgrade

To ensure you are using the latest version of Ultralytics YOLO, upgrade your `ultralytics` package along with the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Community Support

Meanwhile, we encourage you to join the Ultralytics community for further support and discussions:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- Detailed discussions on [Discourse](https://community.ultralytics.com)
- Community-driven insights on our [Subreddit](https://reddit.com/r/Ultralytics) 📚

## Verified Environments

If you suspect environment-specific issues, YOLO can be run in any of these verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Finally, for current performance and issues, check our CI pipeline status:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>
This badge indicates the success of all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests which verify operations across various platforms and tasks.

This is an automated response to expedite support, but rest assured, an Ultralytics engineer will review your issue and provide further assistance soon. Thank you for your patience! 😊

**Comment by UranLajciSolaborate** [comment]
Hope to hear an answer to this very important question @UltralyticsAssistant

**Comment by Y-T-G** [comment]
Scale is not just zooming in. But it's also zooming out. So I am note sure if that's reason. It could just be due to randomness unless you ran it multiple times and confirmed that it always brings improvement with smaller `scale`

**Comment by ErnisMeshi** [comment]
Thank you for your quick response. I understand that scaling is used for zooming in and out. What I can say with certainty is that the model's performance significantly improves when I crop the image. For example, after cropping, the image dimensions become (448, 768), matching the model's input size. This suggests that the model is learning to detect narrow objects only when they are zoomed in.
I have conducted additional tests and observed that while performance improves in some cases, the model still fails to detect the object in others.

One potential solution I’m considering is implementing a logic to stop the scaling operation during the final epochs (similar to the close_mosaic parameter). This might help the model generate better results.
Another idea I have in mind is to replace the uniform random distribution for generating scale sizes with a Gaussian distribution.
![image](https://github.com/user-attachments/assets/11034324-8b1e-4e55-901e-324d7595c566)
Please let me know if you have any suggestions to address this issue. I would greatly appreciate your assistance.

**Comment by Y-T-G** [comment]
You can try P2 model for small objects

**Comment by ErnisMeshi** [comment]
That sound interesting. Based on my search of your repo, yolov8-p2.yaml is present but yolo11-p2.yaml is missing. 
Could give me any hint on how to find or write  yolo11-p2.yaml.

**Comment by Y-T-G** [comment]
You can try the one in this PR

https://github.com/ultralytics/ultralytics/pull/16558

**Comment by UranLajciSolaborate** [comment]
Is there something like https://github.com/ultralytics/ultralytics/pull/16558 but for yolo11-pose-P2.yaml

**Comment by Y-T-G** [comment]
You can just change the last layer to Pose
